main.go

```diff
@@ -1320,3 +1317,60 @@ func newGRPCServer(cfg *config, tenantHeader string, tenantIDs map[string]string
 
 	return gs, nil
 }
+
+type groupHandler struct {
+	group   string
+	handler string
+}
+
+var legacyMetricsGroup = map[string]groupHandler{
+	metricslegacy.QueryRoute:      {"metricslegacy", "query"},
+	metricslegacy.QueryRangeRoute: {"metricslegacy", "query_range"},
+}
+
+var metricsV1Group = map[string]groupHandler{
+	metricsv1.UIRoute:          {"metricsv1", "ui"},
+	metricsv1.QueryRoute:       {"metricsv1", "query"},
+	metricsv1.QueryRangeRoute:  {"metricsv1", "query_range"},
+	metricsv1.SeriesRoute:      {"metricsv1", "series"},
+	metricsv1.LabelNamesRoute:  {"metricsv1", "labels"},
+	metricsv1.LabelValuesRoute: {"metricsv1", "labelvalues"},
+	metricsv1.ReceiveRoute:     {"metricsv1", "receive"},
+	metricsv1.RulesRoute:       {"metricsv1", "rules"},
+	metricsv1.RulesRawRoute:    {"metricsv1", "rules"},
+}
```

Comment:
Maybe we can put the `groupHandler` struct in `server` package and the `metricsV1Group` and `legaceMetricsGroup` maps in the `legacy` and `v1` package? Then import them here to use in `chiLabelParser`?

Reply:
I agree with the sentiment but the contents are tied to the input in this package/file on https://github.com/observatorium/api/blob/main/main.go#L509 so if its not a blocker, I suggest we leave as is for now.

I do have some ideas for restructuring though to make it cleaner overall that I can add in a follow on PR

Reply:
Ack! We can iterate on this then.